### ISSUE TITLE
Cancellation support for `Futures::loop`

### ISSUE THREAD

**Issue by EronWright** [pravega/pravega#2258]
**Problem description**
The future that is returned by `Futures::loop` should support cancellation by terminating the loop.   For example, the Flink connector cancels the future produced by `ReaderGroup::initiateCheckpoint`  after some timeout.

**Problem location**
`io.pravega.common.concurrent.Futures`

**Comment by andreipaduroiu** [comment]
Isn't the condition sufficient?
```
public static CompletableFuture<Void> loop(Supplier<Boolean> condition, Supplier<CompletableFuture<Void>> loopBody, Executor executor)
```
If you return `false` when `condition` is executed, the loop will exit.

**Comment by EronWright** [comment]
@andreipaduroiu that is an alright workaround, but it is easy to add first-class support.

**Comment by EronWright** [comment]
Was incorrectly closed by #4102.

**Comment by EronWright** [comment]
Closing due to lack of interest.
